modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -0,0 +1,107 @@
+package org.tessellation.sdk.infrastructure.consensus
+
+import cats.effect.Async
+import cats.kernel.Next
+import cats.syntax.all._
+import cats.{Applicative, Order, Show}
+
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe.TypeTag
+
+import org.tessellation.ext.cats.syntax.next._
+import org.tessellation.sdk.domain.consensus.ConsensusFunctions
+
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+trait ConsensusManager[F[_], Event, Key, Artifact] {
+
+  def checkForTrigger(event: Event): F[Unit]
+  def checkForStateUpdate(key: Key)(resources: ConsensusResources[Artifact]): F[Unit]
+}
+
+object ConsensusManager {
+
+  def make[F[_]: Async, Event, Key: Show: Order: Next: TypeTag: ClassTag, Artifact <: AnyRef: Show: TypeTag](
+    consensusFns: ConsensusFunctions[F, Event, Key, Artifact],
+    consensusStorage: ConsensusStorage[F, Event, Key, Artifact],
+    consensusStateUpdater: ConsensusStateUpdater[F, Key, Artifact]
+  ): ConsensusManager[F, Event, Key, Artifact] =
+    new ConsensusManager[F, Event, Key, Artifact] {
+
+      private val logger = Slf4jLogger.getLoggerFromClass[F](ConsensusManager.getClass)
+
+      def checkForTrigger(event: Event): F[Unit] =
+        consensusStorage.getLastKeyAndArtifact.flatMap(_.traverse(internalForCheckTrigger(_, event)).void)
+
+      def checkForStateUpdate(key: Key)(resources: ConsensusResources[Artifact]): F[Unit] =
+        for {
+          maybeState <- consensusStorage.getState(key)
+          _ <- maybeState.traverse(state => internalCheckForStateUpdate(key, state, resources))
+        } yield ()
+
+      private def internalForCheckTrigger(lastKeyAndArtifact: (Key, Artifact), event: Event): F[Unit] =
+        if (consensusFns.triggerPredicate(lastKeyAndArtifact, event)) {
+          val nextKey = lastKeyAndArtifact._1.next
+          consensusStorage
+            .getState(nextKey)
+            .map(_.isEmpty)
+            .ifM(
+              logger.info(s"Triggering consensus for key ${nextKey.show}") >>
```

Comment:
Idk, I think it's actually a useful log and I would leave it on info. What do you think?

Reply:
I think we can keep in `info` information about selecting global snapshot only, and IMO there is no need to log all rounds or sth (I mean, it will be useful but for debugging)